llama/llama.go

```diff
@@ -82,29 +82,37 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"log"
 	"os"
 	"strings"
-	"time"
+	"sync"
 	"unicode/utf8"
 	"unsafe"
 
 	"github.com/jmorganca/ollama/api"
 )
 
-type llama struct {
+type LLM struct {
 	params *C.struct_llama_context_params
 	model  *C.struct_llama_model
 	ctx    *C.struct_llama_context
 
+	lastN  deque[C.llama_token]
+	embd   []C.llama_token
+	cursor int
+
+	mu sync.Mutex
+	gc bool
```

Comment:
This is the parameter used to stop a loaded model?

Reply:
Yes it's to signal the model to terminate prediction